tests/assert_type/db/models/test_enums.py

```diff
@@ -0,0 +1,132 @@
+import enum
+from typing import Literal, Optional
+
+from django.db.models import Choices, IntegerChoices, TextChoices
+from django.utils.functional import _StrOrPromise
+from django.utils.translation import gettext_lazy as _
+from typing_extensions import assert_type
+
+
+class Suit(IntegerChoices):
+    DIAMOND = 1, _("Diamond")
+    SPADE = 2, _("Spade")
+    HEART = 3, _("Heart")
+    CLUB = 4, _("Club")
+
+
+class YearInSchool(TextChoices):
+    FRESHMAN = "FR", _("Freshman")
+    SOPHOMORE = "SO", _("Sophomore")
+    JUNIOR = "JR", _("Junior")
+    SENIOR = "SR", _("Senior")
+    GRADUATE = "GR", _("Graduate")
+
+
+class Vehicle(IntegerChoices):
+    CAR = 1, "Carriage"
+    TRUCK = 2
+    JET_SKI = 3
+
+    __empty__ = _("(Unknown)")
+
+
+class Gender(TextChoices):
+    MALE = "M"
+    FEMALE = "F"
+    NOT_SPECIFIED = "X"
+
+    __empty__ = "(Undeclared)"
+
+
+class Medal(TextChoices):
+    GOLD = enum.auto()
+    SILVER = enum.auto()
+    BRONZE = enum.auto()
```

Comment:
Do we care about a special case with `auto()` and `__empty__`?

Reply:
I don't think we need it - the two things don't interact.